py/frequenz/client/weather/_types.py

```diff
@@ -263,19 +256,21 @@ def to_ndarray_vlf(
 
             # Check if the array shape matches the number of filtered times, locations
             # and features
-            if array.shape[0] != len(validity_times_indexes):
+            if validity_times is not None and array.shape[0] != len(validity_times):
                 print(
-                    f"Warning:  The count of validity times in the "
-                    f"array({array.shape[0]}) does not match the expected time "
-                    f"filter count ({validity_times_indexes}."
+                    (
```

Comment:
Do we need these extra layers of parantheses? Also, `print`s should be replaced with logger.

Reply:
I would rework the print statements in a different PR, i already wrote an issue for that